File: benchmarks/MergeSort.php
<?php
use function Cases\testMergeSort;

include(__DIR__ . '/Benchmark.php');
include(__DIR__ . '/../cases/merge-sort/MergeSort.php');
include(__DIR__ . '/../cases/merge-sort/MergeSort2.php');

$testingSize = 10000;

$testRandom = [];
for ($i = 0; $i < $testingSize; $i++) {
    $testRandom[] = rand(0, 10000);
}

$testSortedAsc = [];
for ($i = 0; $i < $testingSize; $i++) {
    $testSortedAsc[] = $i;
}

$testSortedDesc = [];
for ($i = $testingSize; $i > 0; $i--) {
    $testSortedDesc[] = $i;
}

echo "Testing MergeSort - Random asc\n";
$tmp = $testRandom;
Benchmark::start('MergeSort - Random asc');
testMergeSort($tmp);
Benchmark::end('MergeSort - Random asc');

echo "Testing MergeSort - Sort already sorted\n";
$tmp = $testSortedAsc;
Benchmark::start('MergeSort - Sort already sorted');
testMergeSort($tmp);
Benchmark::end('MergeSort - Sort already sorted');

echo "Testing MergeSort - Desc sort to asc\n";
$tmp = $testSortedDesc;
Benchmark::start('MergeSort - Desc sort to asc');
testMergeSort($tmp);
Benchmark::end('MergeSort - Desc sort to asc');

echo "Testing MergeSort2 - Random asc\n";
$tmp = $testRandom;
Benchmark::start('MergeSort2 - Random asc');
mergeSort($tmp);
Benchmark::end('MergeSort2 - Random asc');

echo "Testing MergeSort - Sort already sorted\n";
$tmp = $testSortedAsc;
Benchmark::start('MergeSort2 - Sort already sorted');
mergeSort($tmp);
Benchmark::end('MergeSort2 - Sort already sorted');

echo "Testing MergeSort - Desc sort to asc\n";
$tmp = $testSortedDesc;
Benchmark::start('MergeSort2 - Desc sort to asc');
mergeSort($tmp);
Benchmark::end('MergeSort2 - Desc sort to asc');


print_r(Benchmark::getAll());

File: benchmarks/MergeSort.py
# Inhoud van benchmark_merge_sort.py
import json
import sys
import time
import random

sys.path.append('/workspaces/ADP')

from cases.merge_sort.MergeSort import merge_sort

def benchmark(test_name, data):
    start_time = time.time()
    merge_sort(data)
    end_time = time.time()
    print(f"{test_name}: {end_time - start_time} seconden")

testing_size = 10000

test_random = [random.randint(0, 10000) for _ in range(testing_size)]
test_sorted_asc = list(range(testing_size))
test_sorted_desc = list(range(testing_size, 0, -1))

print("Testing MergeSort - Random")
benchmark("MergeSort - Random", test_random.copy())

print("Testing MergeSort - Sort already sorted")
benchmark("MergeSort - Sort already sorted", test_sorted_asc.copy())

print("Testing MergeSort - Desc sort to asc")
benchmark("MergeSort - Desc sort to asc", test_sorted_desc.copy())